feedbooks.py

```diff
@@ -29,23 +32,71 @@ def extract_feed_data(self, feed, feed_url=None):
             self.improve_description(id, m)
         return metadata, failures
 
-    def rights_for_entry(self, entry):
+    @classmethod
+    def rights_uri_from_feedparser_entry(cls, entry):
         """Determine the URI that best encapsulates the rights status of
         the downloads associated with this book.
 
-        Most of the time this will be CC-BY-NC, the rights status
-        associated with Feedbooks, or "full copyright", for books that
-        are redistributable in France but not in the US. In rare cases
-        it will be some other CC license that prohibits Feedbooks from
-        relicensing as CC-BY-NC.
-
-        :return: A URI.
+        This answer is not reliable because we don't actually have
+        access to dcterms:issued, but we do the best we can. This will
+        be called by _detail_for_feedparser_entry and fixed by
+        _detail_for_elementtree_entry.
         """
         rights = entry.get('rights', "")
         source = entry.get('dcterms_source', '')
-        publication_year = entry.get('dcterms_issued', None)
+        issued = entry.get('dcterms_issued', None) # :(
+        return RehostingPolicy.rights_uri(rights, source, issued)
+
+    @classmethod
+    def rights_uri_from_entry_tag(cls, entry):
+        rights = OPDSXMLParser._xpath1(entry, 'atom:rights')
+        if rights is not None:
+            rights = rights.text
+        source = OPDSXMLParser._xpath1(entry, 'dcterms:source')
+        if source is not None:
+            source = source.text
+        publication_year = OPDSXMLParser._xpath1(entry, 'dcterms:issued')
+        if publication_year is not None:
+            publication_year = publication_year.text
```

Comment:
This method is called by code in core OPDSImporter. We extract the information necessary to determine our rights to this book (as opposed to Feedbooks's rights), and run it through the preexisting `RehostingPolicy.rights_uri` method.